src/modality_docs.cpp

```diff
@@ -1387,23 +1438,39 @@ void ukv_docs_read(ukv_docs_read_t* c_ptr) {
     growing_tape_t growing_tape {arena};
     growing_tape.reserve(places.size(), c.error);
     return_on_error(c.error);
+    sj::ondemand::parser parser;
 
-    auto safe_callback = [&](ukv_size_t, ukv_str_view_t field, json_t const& doc) {
-        yyjson_val* root = yyjson_doc_get_root(doc.handle);
-        yyjson_val* branch = json_lookup(root, field);
-        any_dump({.handle = branch}, c.type, arena, growing_tape, c.error);
+    auto safe_callback = [&](ukv_size_t, ukv_str_view_t field, value_view_t binary_doc) {
+        if (binary_doc.empty()) {
+            growing_tape.push_back(binary_doc, c.error);
+            return;
+        }
+        auto maybe_doc = parser.iterate((const uint8_t*)binary_doc.data(),
+                                        binary_doc.size(),
+                                        binary_doc.size() + sj::SIMDJSON_PADDING);
```

Comment:
This may cause SEGFAULT and was already solved in the other parts of the file, by using an additional buffer with padding at the end.